plugins/module_utils/network/sonic/config/mfa/mfa.py

```diff
@@ -0,0 +1,702 @@
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        if want:
+            want = remove_empties(want)
+        have = existing_mfa_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        state = self._module.params['state']
+        want = self.validate_want(want, have, state)
+
+        if state == 'overridden':
+            commands, requests = self._state_overridden(want, have)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced(want, have)
+        return commands, requests
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands, mod_commands, requests = [], [], []
+
+        replaced_config, requests = self.get_replaced_config(want, have)
+        if replaced_config:
+            commands.extend(update_states(replaced_config, "deleted"))
+            mod_commands = want
+        else:
+            diff = get_diff(want, have, TEST_KEYS)
+            mod_commands = diff
+
+        if mod_commands:
+            mod_requests = self.get_modify_mfa_requests(mod_commands, have)
+            if len(mod_requests) > 0:
+                requests.extend(mod_requests)
+                commands.extend(update_states(mod_commands, "replaced"))
+
+        return commands, requests
+
+    def _state_overridden(self, want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands, requests = [], []
+
+        del_commands = get_diff(have, want, TEST_KEYS)
+        self.remove_default_entries(del_commands)
+
+        if del_commands:
+            is_delete_all = True
+            have, del_requests = self.get_delete_mfa_requests(want, have, is_delete_all)
+            requests.extend(del_requests)
+            commands.extend(update_states(have, "deleted"))
+            have = {}
+
+        if not have and want:
```

Comment:
New configuration needs to be applied irrespective of deletion.
```suggestion
        if want:
```